src/app/parcels/batch/AddressEditCell.tsx

```diff
@@ -0,0 +1,79 @@
+import { Popover } from "@mui/material";
+import { GridRenderCellParams, useGridApiContext } from "@mui/x-data-grid";
+import { useEffect, useRef, useState } from "react";
+import { BatchActionType, BatchTableDataState } from "@/app/parcels/batch/BatchTypes";
+import { addressToString } from "@/app/parcels/batch/displayHelpers";
+import AddressEditCellInput from "@/app/parcels/batch/AddressEditCellInput";
+
+interface AddressEditCellProps {
+    gridRenderCellParams: GridRenderCellParams;
+    tableState: BatchTableDataState;
+    dispatchBatchTableAction: React.Dispatch<BatchActionType>;
+}
+
+const AddressEditCell: React.FC<AddressEditCellProps> = ({
+    gridRenderCellParams,
+    tableState,
+    dispatchBatchTableAction,
+}) => {
+    const id = gridRenderCellParams.id as number;
+    const apiRef = useGridApiContext();
+    const [popoverAnchorEl, setPopoverAnchorEl] = useState<null | HTMLElement>(null);
+    const gridCellReference = useRef<HTMLDivElement>(null);
+    const open = Boolean(popoverAnchorEl);
+    useEffect(() => {
+        setPopoverAnchorEl(gridCellReference.current);
+    }, []);
```

Comment:
Is the useEffect and the set function required? As both the popover element and the ref are already defaulted to null

Reply:
Yes, as without it either the anchorEl will always be null or infinite re-renders will occur